calculator.rb

```diff
@@ -0,0 +1,111 @@
+# If the result is a whole number, program should return an integer. Otherwise, it should return a float rounded to 2 decimal points.
+def checkForDecimal(result)
+  if result % 1 == 0
+    result = result.to_i
+  else
+    result = result.to_f.round(2)
+  end
+end
+
+# Methods for each operator
+def add(num1, num2)
+  result = num1 + num2
+  checkForDecimal(result)
+end
+
+def subtract(num1, num2)
+  result = num1 - num2
+  checkForDecimal(result)
+end
+
+def multiply(num1, num2)
+  result = num1 * num2
+  checkForDecimal(result)
+end
+
+def divide(num1, num2)
+  result = num1 / num2
+  checkForDecimal(result)
+end
+
+def exponent(num1, num2)
+  result = num1 ** num2
+  checkForDecimal(result)
+end
+
+def modulo(num1, num2)
+  result = num1 % num2
+  checkForDecimal(result)
+end
+
+# Methods for getting user numbers. Proram should keep prompting user until they enter an integer.
+def getFirstUserNum
+  begin
+    return Float(gets.chomp)
+  rescue
+    puts "Please enter the number as an integer."
+    retry
+  end
+end
+
+def getSecondUserNum
+  begin
+    return Float(gets.chomp)
+  rescue
+    puts "Please enter the number as an integer."
+    retry
+  end
+end
+
+# Program should continuously run until the user manually quits
+continue = " "
+while continue != "q"
+
+  # Ask user for an operation, should accept strings or symbols for each operation supported by the program
+  puts "Welcome to the Ada calculator! Currently our machine can handle the following operations: Add (+), Subtract (-), Multiply (*), Divide (/), Exponent (^), and Modulo (%). Please select one operation (type either the name or the symbol): "
+  operators = %w(add + subtract - multiply * divide / exponent ^ modulo %)
+  operator = gets.chomp.downcase
+  until operators.include?(operator)
+    puts "Please enter one of the following operations: Add (+), Subtract (-), Multiply (*), Divide (/), Exponent (^), or Modulo (%). Please select one operation (type either the name or the symbol): "
+    operator = gets.chomp.downcase
+  end
+
+  # Ask user for two numbers
+  puts "Please enter your first number: "
+  num1 = getFirstUserNum
+
+  puts "Please enter your second number: "
+  num2 = getSecondUserNum
+
+  # If user is dividing numbers, prompt them for different numbers if they attempt to divide something by 0
+  if (operator == "/" || operator == "divide") && num2 == 0
+    puts "Sorry, you can't divide a number by 0. Please re-type the numbers you would like to divide: "
+    num1 = getFirstUserNum
```

Comment:
This should be a `while` loop - if the user enters 0 for the denominator again, then the program won't catch it.